src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3442,15 +3463,17 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             (ImmutableArray<BoundExpression> argumentsNoConversions, ImmutableArray<Conversion> conversions) = RemoveArgumentConversions(arguments, refKindsOpt);
 
             // Visit the arguments and collect results
-            ImmutableArray<VisitArgumentResult> results = VisitArgumentsEvaluate(argumentsNoConversions, refKindsOpt, parameters, argsToParamsOpt, expanded);
+            ImmutableArray<VisitArgumentResult> results;
+            (results, argsToParamsOpt, refKindsOpt) = VisitArgumentsEvaluate(node.Syntax, argumentsNoConversions, refKindsOpt, parametersOpt, argsToParamsOpt, expanded);
 
             // Re-infer method type parameters
             if ((object)method != null && method.IsGenericMethod)
             {
                 if (HasImplicitTypeArguments(node))
                 {
-                    method = InferMethodTypeArguments(node, method, GetArgumentsForMethodTypeInference(argumentsNoConversions, results));
-                    parameters = method.Parameters;
+                    var binder = (node as BoundCall)?.BinderOpt ?? (node as BoundCollectionElementInitializer)?.BinderOpt ?? throw ExceptionUtilities.UnexpectedValue(node);
```

Comment:
>BoundCall [](start = 42, length = 9)

Looking at the different kinds of bound nodes that have `BinderOpt` I noticed `BoundIndexerAccess`. and `BoundObjectCreation`. Should they be handled too?

Reply:
Perhaps, I only did these because the overload of InferMethodTypeArguments that I deleted would switch on these two types and reject any other types.

Reply:
The reason I only handled these two is that the InferMethodTypeArguments overload that I deleted only handled these two.

---
In reply to: [329246636](https://github.com/dotnet/roslyn/pull/38788#discussion_r329246636) [](ancestors = 329246636)

Reply:
Could you add a test with indexer and default parameter?

---
In reply to: [329251239](https://github.com/dotnet/roslyn/pull/38788#discussion_r329251239) [](ancestors = 329251239)

Reply:
I will add the test with the expectation that NotNullIfNotNull will not have an effect.

---
In reply to: [329269700](https://github.com/dotnet/roslyn/pull/38788#discussion_r329269700) [](ancestors = 329269700,329251239)